open_city_profile/graphene.py

```diff
@@ -78,54 +77,106 @@ def resolve(self, next, root, info, **kwargs):
         return next(root, info, **kwargs)
 
 
-HelTranslationLanguageType = graphql.GraphQLEnumType(
-    name="HelTranslationLanguage",
-    values=OrderedDict(
-        {
-            lang[0].upper(): graphql.GraphQLEnumValue(lang[0], description=lang[1])
-            for lang in settings.LANGUAGES
-        }
-    ),
-)
+def _parler_field_resolver(attname, instance, info, language=None):
+    if language:
+        return instance.safe_translation_getter(attname, language_code=language)
 
+    return getattr(instance, attname)
 
-HelTranslationDirective = graphql.GraphQLDirective(
-    name="hel_translation",
-    locations=[graphql.DirectiveLocation.FIELD],
-    args={
-        "in": graphql.GraphQLArgument(
-            graphql.GraphQLNonNull(HelTranslationLanguageType),
-            description="The language",
-        ),
-    },
-    description="Get value translated in language",
-)
 
+def _get_language_choices_from_parler_settings():
+    site_id = getattr(settings, "SITE_ID", None)
+    parler_languages = settings.PARLER_LANGUAGES.get(site_id, [])
 
-class HelTranslationMiddleware:
-    def resolve(self, next_middleware, root, info, **kwargs):
-        directives = info.field_asts[0].directives
+    choices = []
+    for parler_language in parler_languages:
+        language_code = parler_language["code"]
+        choices.append((language_code.upper(), language_code))
 
-        hel_translation_directives = [
-            directive
-            for directive in directives
-            if directive.name.value == "hel_translation"
-        ]
-        if not hel_translation_directives:
-            return next_middleware(root, info, **kwargs)
+    return choices
 
-        directive = hel_translation_directives[0]
-        in_argument = next(arg for arg in directive.arguments if arg.name.value == "in")
-        language = graphql.value_from_ast(
-            in_argument.value, HelTranslationLanguageType, info.variable_values
-        )
 
-        if isinstance(root, TranslatableModel):
-            # switch_language will change Django language too
-            with switch_language(root, language):
-                return_value = next_middleware(root, info, **kwargs)
-        else:
-            with translation.override(language):
-                return_value = next_middleware(root, info, **kwargs)
+TranslationLanguage = graphene.Enum(
+    "TranslationLanguage", _get_language_choices_from_parler_settings()
+)
+
 
-        return return_value
+class DjangoParlerObjectType(DjangoObjectType):
+    class Meta:
+        abstract = True
+
```

Comment:
It's a Django feature. Has to be abstract because `DjangoObjectType` has a bit of an [issue](https://github.com/graphql-python/graphene-django/issues/423) when trying to subclass it.

Reply:
Aha. It seems to be a graphene-django feature in this case. But an undocumented one, as are so many other features in that library :facepalm: